test/benchmarks/src/main/scala/scala/collection/mutable/OpenHashMapBenchmark.scala

```diff
@@ -0,0 +1,204 @@
+    def iterationTeardown(params: IterationParams) {
+      if (params.getType == IterationType.MEASUREMENT) {
+        // limit to smaller cases to avoid OOM
+        memory = if (mapEntries <= 1000000) GraphLayout.parseInstance(maps(0), maps.tail).totalSize else 0
+      }
+    }
+  }
+
+  /** State container for the `get()` bulk calling tests.
+    * 
+    * Provides a thread-scoped map of the expected size.
+    * Performs a GC after loading the map.
+    */
+  @State(Scope.Thread)
+  class BulkGetState {
+    val map = new OpenHashMap[Int,Int].empty
+
+    /** Load the map with keys from `1` to `size`. */
+    @Setup
+    def setup(params: BenchmarkParams) {
+      val size = params.getParam("size").toInt
+      put_Int(map, 1, size)
+      System.gc()
+    }
+  }
+
+  /** State container for the `get()` bulk calling tests with deleted entries.
+    * 
+    * Provides a thread-scoped map of the expected size, from which entries have been removed.
+    * Performs a GC after loading the map.
+    */
+  @State(Scope.Thread)
+  class BulkRemovedGetState {
+    val map = new OpenHashMap[Int,Int].empty
+
+    /** Load the map with keys from `1` to `size`, removing half of them. */
+    @Setup
+    def setup(params: BenchmarkParams) {
+      val size = params.getParam("size").toInt
+      put_remove_Int(map, size)
+      System.gc()
+    }
+  }
+
+  /** Put elements into the given map. */
+  private def put_Int(map: OpenHashMap[Int,Int], from: Int, to: Int) {
+    var i = from
+    while (i <= to) {  // using a `for` expression instead adds significant overhead
+      map.put(i, i)
+      i += 1
+    }
+  }
+
+  /** Put elements into the given map, removing half of them as they're added.
+    * 
+    * @param size number of entries to leave in the map on return
+    */
+  def put_remove_Int(map: OpenHashMap[Int,Int], size: Int) {
+    val blocks = 50  // should be a factor of `size`
+    val totalPuts = 2 * size  // add twice as many, because we remove half of them
+    val blockSize: Int = totalPuts / blocks
+    var base = 0
+    while (base < totalPuts) {
+      put_Int(map, base + 1, base + blockSize)
+
+      // remove every other entry
+      var i = base + 1
+      while (i <= base + blockSize) {
+        map.remove(i)
+        i += 2
+      }
+
+      base += blockSize
+    }
+  }
+
+  /** Get elements from the given map. */
+  def get_Int(map: OpenHashMap[Int,Int], size: Int, bh: Blackhole) {
+    var i = 1
+    while (i <= size) {
+      bh.consume(map.get(i).getOrElse(0))
+      i += 1
+    }
+  }
+}
+
+/** Benchmark for the library's [[OpenHashMap]]. */
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(6)
+@Threads(1)
+@Warmup(iterations = 20)
+@Measurement(iterations = 6)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Benchmark)
+class OpenHashMapBenchmark {
+  import OpenHashMapBenchmark._
+
+  @Param(Array("50", "100", "250", "1000", "2500", "10000", "25000", "100000", "250000", "1000000", "2500000",
```

Comment:
Unless you declare that a map is only to be used for big collections, it's also valuable to test the smaller sizes.  This generally involves making an array of maps and doing something to each of them, as there is too much overhead to measure accurately the timings in a map of e.g. 2 elements.  You can just run all your benchmarks that way, since it's perfectly okay to have an array of length 1.